chunk_size should always be int64_t for Foreach functors

janeyx99 opened the pull request:
See https://github.com/pytorch/pytorch/issues/156261#issuecomment-3002394773

Testing is a valid q--it is pretty expensive to test such large tensors for all these ops.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #156872
* #156871
* #156876

janeyx99 committed:
chunk_size should always be int64_t for Foreach functors

[ghstack-poisoned]

ngimel commented:
I'm not sure if all foreach functors strictly require this, but probably it doesn't hurt.

janeyx99 committed:
Update on "chunk_size should always be int64_t for Foreach functors"

See https://github.com/pytorch/pytorch/issues/156261#issuecomment-3002394773

Testing is a valid q--it is pretty expensive to test such large tensors for all these ops.




[ghstack-poisoned]

albanD reviewed (commented):
I see you're not changing any caller into these operators? Does that mean that we used to have a warning somewhere that we were narrowing the argument we ignored?

janeyx99 committed:
Update on "chunk_size should always be int64_t for Foreach functors"

See https://github.com/pytorch/pytorch/issues/156261#issuecomment-3002394773

Testing is a valid q--it is pretty expensive to test such large tensors for all these ops.




[ghstack-poisoned]

janeyx99 commented:
> I see you're not changing any caller into these operators? Does that mean that we used to have a warning somewhere that we were narrowing the argument we ignored?

We don't have a warning, but these functors are called here, where kChunkSize is an int64_t: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cuda/MultiTensorApply.cuh#L106
